source/adios2/core/ADIOS.cpp

```diff
@@ -79,6 +79,18 @@ IO &ADIOS::DeclareIO(const std::string ioName)
     return ioPair.first->second;
 }
 
+IO &ADIOS::GetIO(const std::string name)
+{
+    auto itIO = m_IOs.find(name);
+    if (itIO == m_IOs.end())
```

Comment:
the STL will throw its own exception if itIO is the end iterator.  Then then user will have to turn on m_DebugMode (which is the recommended mode for development) if ADIOS catches this and provides more context.  That's how I am handling all exception checks. For well established flows it's unnecessary to do the same checks. Also, we are providing the option to other library developers to handle exceptions their own way and not always from adios.